tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Http/WebRequest/HeadersInjectedCache.cs

```diff
@@ -0,0 +1,31 @@
+// <copyright file="HeadersInjectedCache.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System.Net;
+using System.Runtime.CompilerServices;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Http.WebRequest;
+
+internal static class HeadersInjectedCache
+{
+    private static readonly object InjectedValue = new();
+    private static readonly ConditionalWeakTable<WebHeaderCollection, object> Cache = new();
```

Comment:
I'm starting to think you're a shill for "Big ConditionalWeakTable"🕵️‍♂️

Reply:
It's too late for me, I can't turn back now 😂